app/forms/work_form.rb

```diff
@@ -0,0 +1,95 @@
+# typed: false
+# frozen_string_literal: true
+
+require 'reform/form/coercion'
+
+class WorkForm < Reform::Form
+  feature Coercion
+
+  property :work_type
+  property :subtype, default: 'manuscript'
+  property :title
+  property :contact_email
+  property :abstract
+  property :citation
+  property :collection_id
+  property :license
+  property :agree_to_terms
+
+  property 'created(1i)', virtual: true
+  property 'created(2i)', virtual: true
+  property 'created(3i)', virtual: true
+  property 'created_range(1i)', virtual: true
+  property 'created_range(2i)', virtual: true
+  property 'created_range(3i)', virtual: true
+  property 'created_range(4i)', virtual: true
+  property 'created_range(5i)', virtual: true
+  property 'created_range(6i)', virtual: true
+  property 'published(1i)', virtual: true
+  property 'published(2i)', virtual: true
+  property 'published(3i)', virtual: true
+  property :creation_type, virtual: true, default: 'single'
+
+  def sync(*)
+    model.created_edtf = case creation_type
+                         when 'range'
+                           deserialize_edtf_range(:created_range)
+                         else
+                           deserialize_edtf(:created)
+                         end
+
+    model.published_edtf = deserialize_edtf(:published)
+
+    super
+  end
+
+  validates :title, presence: true
+
+  collection :contributors,
+             populator: lambda { |fragment:, **|
+               # find out if incoming Contributor is already added.
+               item = contributors.find_by(id: fragment['id']) if fragment['id'].present?
+
+               if fragment['_destroy'] == '1'
+                 contributors.delete(item)
+                 return skip!
+               elsif fragment['first_name'].blank?
+                 return skip!
+               end
+               item || contributors.append(Contributor.new)
+             } do
```

Comment:
What is the fragment here? Even though I've used reform a bit in the past, this is a bit inscrutable to me. Add a code comment?

Reply:
Fragement is the row data from the html form.

Reply:
@jcoyne sorry to be a broken record, but... can you add a comment about that? I don't think our collective reform knowledge is up to snuff.

Reply:
It's added.